test.py

```diff
@@ -5,71 +5,84 @@
-
-# class Test_crop():
-#
-#     def test_crop_shape(self):
-#         '''
-#         Function to test that the crop() function returns an image
-#         of the correct shape
-#         '''
-#         # Test for correct shape
-#         assert util.crop(image, 15, 10).shape == (15, 10, 3)
-#         assert util.crop(image, 10, 5).shape == (10, 5, 3)
-#         assert util.crop(image, 6, 6).shape == (6, 6, 3)
-#         assert util.crop(image, 5, 5).shape == (5, 5, 3)
-#         assert util.crop(image, 1, 1).shape == (1, 1, 3)
-#         assert util.crop(image, image.shape[0], image.shape[1]).shape == image.shape
-#
-#     def test_crop_value(self):
-#         '''
-#         Function to test that ValueErrors are raised when invalid width and
-#         height are entered
-#         '''
-#         with pytest.raises(ValueError):
-#             util.crop(image, -1, 10)
-#         with pytest.raises(ValueError):
-#             util.crop(image, 10, -1)
-#         with pytest.raises(ValueError):
-#             util.crop(image, 0, 0)
-#         with pytest.raises(ValueError):
-#             util.crop(image, 0, 10)
-#         with pytest.raises(ValueError):
-#             util.crop(image, 10, 0)
-#         with pytest.raises(ValueError):
-#             util.crop(image, image.shape[0] + 1, 10)
-#         with pytest.raises(ValueError):
-#             util.crop(image, 10, image.shape[1] + 1)
-#         with pytest.raises(ValueError):
-#             util.crop(image, int(-1e30), 10)
-#         with pytest.raises(ValueError):
-#             util.crop(image, 10, int(-1e30))
-#
-#     def test_crop_type(self):
-#         '''
-#         Function to test that a TypeError is raised when the
-#         wrong type of input is passed into the function.
-#         '''
-#         with pytest.raises(TypeError):
-#             util.crop(image, 9.5, 10)
-#         with pytest.raises(TypeError):
-#             util.crop(image, 10, 9.5)
-#         with pytest.raises(TypeError):
-#             util.crop(image, 9.5, 9.5)
-#         with pytest.raises(TypeError):
-#             util.crop(image, -9.9, -4.5)
-#         with pytest.raises(TypeError):
-#             util.crop(image, "10", True)
-#         with pytest.raises(TypeError):
-#             util.crop("image.jpg", 10, 10)
-#
-#     def test_crop_input_shape(self):
-#         '''
-#         Test that correct image has correct shape
-#         '''
-#         assert np.size(image.shape) == 3
+# images for testing
+test_images = ["data/test_image_1717.png", "data/test_image_1718.png", "data/test_image_1816", "data/test_image_1819" ]
+img_path = ["data/test_image_1718"]
+
+class Test_crop():
+
+    def test_crop_shape(self):
+        '''
+        Function to test that the crop() function returns an image
+        of the correct shape for different input shapes
+        '''
+        for image in test_images:
+
+            cropped_image = crop(image, 15, 10)
+            assert imread(cropped_image).shape == (15, 10, 3)
```

Comment:
I think we are not restricting the channels to be equal to 3. The third dimension should not be check explicitly.
This applies to all the tests in this function except the last one.

Reply:
Fixed this, will send another pull request soon